target/bin/addmailuser

```diff
@@ -35,31 +35,41 @@ function __usage
 
 [[ ${1:-} == 'help' ]] && { __usage ; exit 0 ; }
 
-USER="${1}"
+FULL_EMAIL="${1}"
 shift
 PASSWD="${*}"
 
-[[ -z ${USER} ]] && { __usage ; errex 'No username specified' ; }
-[[ "${USER}" =~ .*\@.* ]] || { __usage ; errex 'Username must include the domain' ; }
+[[ -z ${FULL_EMAIL} ]] && { __usage ; errex 'No username specified' ; }
+[[ "${FULL_EMAIL}" =~ .*\@.* ]] || { __usage ; errex 'Username must include the domain' ; }
 
 # Protect config file with lock to avoid race conditions
 touch "${DATABASE}"
-(
-  flock -e 200
-
-  if grep -qi "^$(escape "${USER}")|" "${DATABASE}" 2>/dev/null
-  then
-    echo "User '${USER}' already exists."
-    exit 1
-  fi
-
-  if [[ -z ${PASSWD} ]]
-  then
-    read -r -s -p "Enter Password: " PASSWD
-    echo
-    [[ -z ${PASSWD} ]] && errex "Password must not be empty"
-  fi
-
-  HASH="$(doveadm pw -s SHA512-CRYPT -u "${USER}" -p "${PASSWD}")"
-  echo "${USER}|${HASH}" >> "${DATABASE}"
-) 200< "${DATABASE}"
+create_lock "$(basename "$0")"
+if grep -qi "^$(escape "${FULL_EMAIL}")|" "${DATABASE}" 2>/dev/null
```

Comment:
`2>/dev/null`
Why? Is this, when $DATABASE does not exist yet or what should be catched here?

Reply:
Probably. Although this should never happen, since there is always a `touch $DATABASE` beforehand, right?

Reply:
Hey guys, so that line was actually already in the script before I modified it for the lock. Not sure I can answer your question :(

Reply:
No problem. But every time something is changed, I try to take the opportunity to review the whole file. If we all agree/think it's not necessary, i would remove it.

Reply:
@casperklein awesome, so `if grep -qi "^$(escape "${FULL_EMAIL}")|" "${DATABASE}" 2>/dev/null` now becomes `if grep -qi "^$(escape "${FULL_EMAIL}")|" "${DATABASE}"` ?

Reply:
Correct.